packages/sit-onyx/src/components/OnyxTabs/types.ts

```diff
@@ -0,0 +1,36 @@
+import type { createTabs } from "@sit-onyx/headless";
+import type { InjectionKey, Ref } from "vue";
+import type { DensityProp } from "../../composables/density";
+
+export type OnyxTabsProps<TValue extends PropertyKey = PropertyKey> = DensityProp & {
+  /**
+   * Label of the tabs. Needed for accessibility / screen readers.
+   */
+  label: string;
+  /**
+   * Currently active tab.
+   */
+  modelValue: TValue;
+  /**
+   * If `true`, the tabs will be stretched to the full available width.
+   */
+  stretched?: boolean;
+};
+
+/**
+ * Injection key to provide relevant context data from the tabs parent to the
+ * individual child tab components.
+ */
+export const TABS_INJECTION_KEY = Symbol() as InjectionKey<{
+  /**
+   * Headless composable provided for child tabs for correct accessibility and behavior.
+   */
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any -- needed to fix type error when using generics inside the OnyxTabs
+  headless: ReturnType<typeof createTabs<any>>;
```

Comment:
```suggestion
  headless: ReturnType<typeof createTabs<PropertyKey>>
```

Reply:
I added any because everything else causes a type issue due to the generics in OnyxTabs.vue. Do you have an idea how to solve this?

Reply:
Yes, you need to pass through the generic "TValue" type from the OnyxTab component.
To achieve that you can wrap the provided value of `TABS_INJECTION_KEY` into a function, which can inject the generic.